holidays/countries/nauru.py

```diff
@@ -0,0 +1,102 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from datetime import date
+from gettext import gettext as tr
+
+from holidays.constants import PUBLIC
+from holidays.groups import ChristianHolidays, InternationalHolidays
+from holidays.observed_holiday_base import ObservedHolidayBase, SAT_SUN_TO_NEXT_WORKDAY
+
+
+class Nauru(ObservedHolidayBase, ChristianHolidays, InternationalHolidays):
+    """Nauru holidays.
+
+    References:
+        * <https://www.worldatlas.com/articles/what-languages-are-spoken-in-nauru.html>
+        * <https://publicholidays.asia/nauru/>
+        * <https://commonwealthchamber.com/day-of-the-tribes-nauru/>
+        * <https://anydayguide.com/calendar/2514>
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Nauru>
+        * <https://www.timeanddate.com/holidays/nauru/>
+    """
+
+    country = "NR"
+    default_language = "en_AU"
+    supported_categories = (PUBLIC,)
+    # %s (observed).
+    observed_label = tr("%s (observed)")
+    supported_languages = ("en_AU",)
+    # Nauru gained independence on January 31, 1968
+    start_year = 1969
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        super().__init__(*args, **kwargs)
+
+    def _populate_observed(self, dts: set[date], multiple: bool = False) -> None:
+        for dt in sorted(dts):
+            self._add_observed(dt, rule=SAT_SUN_TO_NEXT_WORKDAY)
+
+    def _populate_public_holidays(self):
+        dts_observed = set()
+
+        # New Year's Day.
+        dts_observed.add(self._add_new_years_day(tr("New Year's Day")))
```

Comment:
We can probably use `self._add_observed( ... )` in place of all `dts_observed.add( ... )` here instead (the latter's usually only required for cases with complex observance i.e. Islamic Holidays overlap with Gregorian ones)
```suggestion
        # New Year's Day.
        self._add_observed(self._add_new_years_day(tr("New Year's Day")))
```